util/vendor.py

```diff
@@ -340,46 +358,82 @@ def main(argv):
         raise SystemExit(sys.exc_info()[1])
     desc['_base_dir'] = vendor_file_base_dir
 
+    first_import = False
+
     # Load lock file contents (if possible)
-    desc_file_path_str = str(desc_file_path)
-    lock_file_path = Path(
-        desc_file_path_str[:desc_file_path_str.find('.vendor.hjson')] +
-        '.lock.hjson')
+    desc_file_stem = desc_file_path.name.rsplit('.', 2)[0]
+    lock_file_path = desc_file_path.with_name(desc_file_stem + '.lock.hjson')
     try:
-        with open(lock_file_path, 'r') as f:
+        with open(str(lock_file_path), 'r') as f:
             lock = hjson.loads(f.read(), use_decimal=True)
     except FileNotFoundError:
-        log.warning(
-            "Unable to read lock file %s. Assuming this is the first import.",
-            lock_file_path)
         lock = None
+        if args.update:
+            log.fatal("Cannot update without existing lock file %s.",
+                      str(lock_file_path))
+            log.fatal(
+                "Retry without the '--update' flag if this is an initial import."
+            )
+            raise SystemExit(1)
+        else:
+            log.warning(
+                "Unable to read lock file %s. Assuming this is the first import.",
+                str(lock_file_path))
+            first_import = True
 
     if args.refresh_patches:
         refresh_patches(desc)
 
     clone_dir = Path(tempfile.mkdtemp())
     try:
+        if first_import or args.update:
+            url_to_clone = desc['upstream']['url']
+            rev_to_clone = desc['upstream']['rev']
+        else:
+            url_to_clone = lock['upstream']['url']
+            rev_to_clone = lock['upstream']['rev']
```

Comment:
only_subdir is also part of the lock file and needs to be used from there.

There is room for discussion if things like the excludes and the applied patches should be part of the lock file as well and applied from there, but that's probably beyond this PR.

To prepare for that, I'd for something like

```
import_desc = desc.copy()
import_desc['upstream'] = lock['upstream']
```

and use import_desc in the code below.

Reply:
I wasn't sure about `only_subdir`, but I think this change does make the code clearer, so I've gone with it. It helped make the documentation clearer too.